includes/Core/Modules/Module.php

```diff
@@ -645,6 +652,31 @@ protected function setup_client() {
 	 */
 	abstract protected function setup_services( Google_Client $client );
 
+	/**
+	 * Sets whether or not to return raw requests and returns a callback to reset to the previous value.
+	 *
+	 * This works similar to e.g. a higher-order component in JavaScript.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param bool $defer Whether or not to return raw requests.
+	 * @return callable Callback function that resets to the original $defer value.
+	 */
+	protected function with_client_defer( $defer ) {
+		$client = $this->get_client();
+		if ( $client instanceof Google_Site_Kit_Client ) {
+			return $client->withDefer( $defer );
+		}
+
+		// "Poly-fill" if not a Site Kit client instance.
```

Comment:
Is this necessary? I understand the reasoning to support flexibility in allowing for different clients for different modules. Why not require that `get_client` always returns an instance of `Google_Site_Kit_Client` instead of `Google_Client`? Won't 3rd party clients likely need to extend the proxy client anyways?

Reply:
I initially thought it'd be nice to keep the original compatibility, but in the long run it's probably better to require our own base client, so I'll update that as suggested.